docs/docs/api_reference.md

```diff
@@ -0,0 +1,203 @@
+---
+slug: /api/
+sidebar_position: 4
+---
+
+# API Reference
+
+## CLI
+
+Task command line tool has the following syntax:
+
+```bash
+task [--flags] [tasks...] [-- CLI_ARGS...]
+```
+
+:::tip
+
+If `--` is given, all remaning arguments to assigned to a special `CLI_ARGS`
+variable
+
+:::
+
+| Short | Flag | Type | Default | Description |
+| - | - | - | - | - |
+| `-c` | `--color` | `bool` | `true` | Colored output. Enabled by default. Set flag to `false` or use `NO_COLOR=1` to disable. |
+| `-C` | `--concurrency` | `int` | | Limit number tasks to run concurrently. |
+| `-d` | `--dir` | `string` | | Sets directory of execution. |
+|      | `--dry` | `bool` | | Compiles and prints tasks in the order that they would be run, without executing them. |
+| `-f` | `--force` | `bool` | | Forces execution even when the task is up-to-date. |
+| `-h` | `--help` | `bool` | | Shows Task usage. |
+| `-i` | `--init` | `bool` | | Creates a new Taskfile.yaml in the current folder. |
+| `-l` | `--list` | `bool` | | Lists tasks with description of current Taskfile. |
+| `-a` | `--list-all` | `bool` | | Lists tasks with or without a description. |
+| `-o` | `--output` | `string` | | Sets output style: [`interleaved`/`group`/`prefixed`]. |
+|      | `--output-group-begin` | `string` | | Message template to print before a task's grouped output. |
+|      | `--output-group-end `  | `string` | | Message template to print after a task's grouped output. |
+| `-p` | `--parallel` | `bool` | | Executes tasks provided on command line in parallel. |
+| `-s` | `--silent` | `bool` | | Disables echoing. |
+|      | `--status` | `bool` | | Exits with non-zero exit code if any of the given tasks is not up-to-date. |
+|      | `--summary` | `bool` | | Show summary about a task. |
+| `-t` | `--taskfile` | `string` | | Choose which Taskfile to run. Defaults to `Taskfile.yml`. |
+| `-v` | `--verbose` | `bool` | | Enables verbose mode. |
+|      | `--version` | `bool` | | Show Task version. |
+| `-w` | `--watch` | `bool` | | Enables watch of the given task. |
+
+## Schema
+
+### Taskfile
+
+| Attribute | Type | Default | Description |
+| - | - | - | - |
+| `version` | `string` | | Version of the Taskfile. The current version is `3`. |
+| `includes` | [`map[string]Include`](#include) | | Additional Taskfiles to be included. |
+| `output` | `string` | `interleaved` | Output mode. Available options: `interleaved`, `group` and `prefixed`. |
+| `method` | `string` | `checksum` | Default method in this Taskfile. Can be overriden in a task by task basis. Available options: `checksum`, `timestamp` and `none`. |
+| `silent` | `bool` | `false` | Default "silent" options for this Taskfile. If `false`, can be overidden with `true` in a task by task basis. |
+| `run` | `string` | `always` | Default "run" option for this Taskfile. Available options: `always`, `once` and `when_changed`. |
+| `vars` | [`map[string]Variable`](#variable) | | Global variables. |
```

Comment:
should the default here be either `nil` or `{}` ?

Reply:
I think we probably don't need to specify when the value would be empty.